presentation/src/main/java/org/sagebionetworks/assessmentmodel/presentation/AssessmentViewModel.kt

```diff
@@ -32,10 +31,39 @@ open class AssessmentViewModel(
         assessmentNodeState.goBackward()
     }
 
+    fun skip() {
+        //Clear current results before going to next node
+        (assessmentNodeState.currentChild?.currentResult as? AnswerResult)?.jsonValue = null
+        goForward()
+    }
+
+    fun goToNode(nodeIdentifier: String) {
+        var node: Node? = null
+        if (ReservedNavigationIdentifier.Beginning.matching(nodeIdentifier)) {
+            node = (assessmentNodeState.node as? NodeContainer)?.children?.firstOrNull()
+        } else {
+            node = (assessmentNodeState.node as? NodeContainer)?.children?.firstOrNull { it.identifier == nodeIdentifier }
+        }
+        node?.let {
+            (assessmentNodeState as? BranchNodeStateImpl)?.moveTo(NavigationPoint(node, assessmentNodeState.currentResult))
+        }
+    }
+
     fun cancel() {
         assessmentNodeState.exitEarly(FinishedReason.Incomplete(saveResult = SaveResults.Never, markFinished = false, declined = false))
     }
 
+    fun continueLater() {
+        /* TODO: Need to provide flag to containing app to save partial progress if applicable -nbrown 4/21/2022
+         * Or just have Cancel method and containing app is responsible for saving partial progress if it wants to.
+         */
+        assessmentNodeState.exitEarly(FinishedReason.Incomplete(saveResult = SaveResults.Never, markFinished = false, declined = false))
```

Comment:
What's the difference between "continueLater" and "cancel"?

Reply:
Originally I was thinking we would need distinguish between a user hitting the x button to close a survey, versus the continue later button. I don't think we care at this level, the containing app and BridgeClientKMM can decide whether or not to save progress if a user exits early. It doesn't matter how they exited early.